gui/src/main/java/org/verapdf/processor/ProcessorImpl.java

```diff
@@ -0,0 +1,276 @@
+			processingResult.setErrorInValidation();
+			processingResult.setErrorInMetadataFixer();
+			processingResult.setErrorInFeatures();
+			processingResult.getErrorMessages().add(
+					"Invalid password for reading encrypted PDF file: " + e.getMessage());
+			return processingResult;	//TODO: do we return here?
+		} catch (IOException e) {
+			LOGGER.error("Error: " + fileDetails.getName() + " is not a PDF format file.", e);
+			processingResult.setErrorInValidation();
+			processingResult.setErrorInMetadataFixer();
+			processingResult.setErrorInFeatures();
+			processingResult.getErrorMessages().add(
+					"Error in reading PDF file: " + e.getMessage());
+			return processingResult; 	//TODO: do we return here?
+		}
+		long endTimeOfValidation = System.currentTimeMillis();
+
+		try {
+			switch (config.getReportType()) {
+				case TEXT:
+					if (validationResult != null) {
+						String reportSummary = (validationResult.isCompliant() ?
+								"PASS " : "FAIL ") + fileDetails.getName();
+						reportOutputStream.write(reportSummary.getBytes());
+						if (config.isVerboseCli()) {
+							Set<RuleId> ruleIds = new HashSet<>();
+							for (TestAssertion assertion : validationResult.getTestAssertions()) {
+								if (assertion.getStatus() == TestAssertion.Status.FAILED) {
+									ruleIds.add(assertion.getRuleId());
+								}
+							}
+							for (RuleId id : ruleIds) {
+								String reportRuleSummary = id.getClause() + "-" + id.getTestNumber();
+								reportOutputStream.write(reportRuleSummary.getBytes());
+							}
+						}
+					}
+					break;
+				case MRR:
+				case HTML:
+					MachineReadableReport machineReadableReport = MachineReadableReport.fromValues(
+							fileDetails,
+							validator == null ? Profiles.defaultProfile()
+									: validator.getProfile(), validationResult,
+							config.isShowPassedRules(),
+							config.getMaxNumberOfDisplayedFailedChecks(),
+							fixerResult, featuresCollection,
+							endTimeOfValidation - startTimeOfValidation);
+					if (config.getReportType() == FormatOption.MRR) {
+						MachineReadableReport.toXml(machineReadableReport,
+								reportOutputStream, Boolean.TRUE);
+					} else if (config.getReportType() == FormatOption.HTML) {
+						File tmp = File.createTempFile("verpdf", "xml");
+						tmp.deleteOnExit();
+						try (OutputStream os = new FileOutputStream(tmp)) {
+							MachineReadableReport.toXml(
+									machineReadableReport, os, Boolean.FALSE);
+						}
+						try (InputStream is = new FileInputStream(tmp)) {
+							HTMLReport.writeHTMLReport(is,
+									reportOutputStream, config.getProfileWikiPath());
+						}
+					}
+					break;
+				case XML:
+					CliReport report = CliReport.fromValues(fileDetails, validationResult,
+							FeaturesReport.fromValues(featuresCollection));
+					CliReport.toXml(report, reportOutputStream, Boolean.TRUE);
+					break;
+				default:
+					//TODO: do we need to do something here?
```

Comment:
throw an illegal state exception. we can get here only in case of null report type or if it will be extended without fixing that logic. As this type enum created for this logic - we have not crush here in last case